### PR TITLE
JS: Don't handle this as global value

### PR DESCRIPTION
**Pull request by matthiasblaesing**
JS allows method calls for all functions, which in turn makes this
available in nearly all contexts. It is thus not sensible to report
this as an undeclared global variable.

There are two visible problems:

image

- `this` is reported as an undeclared global variable
- the `this.a.b` assignment generates a global `this` variable with the object structure

Both issues are fixed by this PR.

Closes: https://github.com/apache/netbeans/issues/4568
Closes: https://github.com/apache/netbeans/issues/4213

### CODE REVIEW

**Comment by matthiasblaesing** [comment]
Before this PR the PRs #5786 and #5796 should be merged.

File: webcommon/javascript2.model/src/org/netbeans/modules/javascript2/model/api/ModelUtils.java [hunk 1]
```diff
@@ -93,8 +93,15 @@ public static JsObjectImpl getJsObject (ModelBuilder builder, List<Identifier> f
             return null;
         }
         JsObject result = builder.getCurrentObject();
-        JsObject tmpObject = null;
         String firstName = fqName.get(0).getName();
+        JsObject tmpObject;
+
+        if("this".equals(firstName)) {
```
Comment (junichi11) [inline webcommon/javascript2.model/src/org/netbeans/modules/javascript2/model/api/ModelUtils.java:99]: Nitpick: formatting, NOI18N if possible

**Review by junichi11 — APPROVED** [review]
Looks good. Thank you!

**Comment by matthiasblaesing** [comment]
@junichi11 Thank you again for the review, especially your good eye. Indeed there is already a constant `THIS` defined (and `// NOI18N`ed), that can be used. I pushed an update.
@qmegas thank you for the test.

**Comment by matthiasblaesing** [comment]
Tests are green, tested by issue reported. Will merge, when #5796 is merged.

**Comment by junichi11** [comment]
> Indeed there is already a constant `THIS` defined (and `// NOI18N`ed), that can be used.

Nice👍

**Pull request merged**